### DISCUSSION TITLE
Bundling and publishing extensions as npm packages

### DISCUSSION THREAD

**Discussion by Abdallatif** [CodinGame/monaco-vscode-api#491] [Q&A]
Are there any docs that explain how to bundle an extension and publish it as an npm package in the same way you did with the default extensions? Is that possible?

I'm trying to use the [vscode-js-debug](https://github.com/microsoft/vscode-js-debug) extension and bundle it as part of my web app. However, it seems like it wasn't designed to work with VSCode Web. I understand that it's possible to have the adapter on the server side, as shown in the demo, but I want to keep the adapter on the client side and attach the VSCode debugger to a running Node.js process.

I understand if this issue might be outside the project's scope so feel free to close it.

**Comment by CGNonofr** [comment]
> Are there any docs that explain how to bundle an extension and publish it as an npm package in the same way you did with the default extensions? Is that possible?

2 rollup plugins are published, they can be used to transform a vsix file or a directory to an importable extension

> I'm trying to use the [vscode-js-debug](https://github.com/microsoft/vscode-js-debug) extension and bundle it as part of my web app. However, it seems like it wasn't designed to work with VSCode Web. I understand that it's possible to have the adapter on the server side, as shown in the demo, but I want to keep the adapter on the client side and attach the VSCode debugger to a running Node.js process.

If you want you client to be attached to the adapter, you'll need a channel between them. That is exactly what is demoed in the demo here. But you wont be able to reuse the `vscode-js-debug` to do that.

An alternative is to use the extension host server, and run the debug adapter server side
